Source/SwiftLintBuiltInRules/Rules/Style/TrailingClosureRule.swift

```diff
@@ -77,6 +118,39 @@ private extension TrailingClosureRule {
     }
 }
 
+private extension TrailingClosureRule {
+    final class Rewriter: ViolationsSyntaxRewriter<ConfigurationType> {
+        override func visit(_ node: FunctionCallExprSyntax) -> ExprSyntax {
+            guard node.trailingClosure == nil else { return super.visit(node) }
+
+            if configuration.onlySingleMutedParameter {
+                if let param = node.singleMutedClosureParameter,
+                let converted = node.convertToTrailingClosure() {
+                     correctionPositions.append(param.positionAfterSkippingLeadingTrivia)
+                    return ExprSyntax(converted)
+                }
+            } else if let param = node.lastDistinctClosureParameter,
+                      let converted = node.convertToTrailingClosure() {
+                correctionPositions.append(param.positionAfterSkippingLeadingTrivia)
+                return ExprSyntax(converted)
```

Comment:
Here and in line 130, we need to call `super.visit` on the rewritten node to handle nested function calls as well. An example would be:

```swift
f(a: { g(b: { 1 }) })
```

I don't think this works currently.

Reply:
Fixed at d14ed77

Reply:
Please add a test for this case.